vignettes/python-primer.Rmd

```diff
@@ -0,0 +1,920 @@
+---
+title: "Primer on Python for R Users"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Primer on Python for R Users}
+  %\VignetteEncoding{UTF-8}
+  %\VignetteEngine{knitr::rmarkdown}
+editor_options: 
+  markdown: 
+    wrap: 72
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+```{r setup}
+library(reticulate)
+```
+
+## Primer on Python for R users
+
+R is the common language of statistics and data analysis, but it is not
+the only language on the scene. You may find yourself occasionally
+wanting to read and understand some Python, or even port some Python to
+R. This guide is designed to enable you to do these tasks as quickly as
+possible. As you'll see, R and Python are similar enough that it's
+possible to do these tasks, without necessarily learning all of Python.
+
+### Whitespace
+
+Whitespace matters in Python. In R you groups expressions into a code
+block with `{}`, in Python you do that by making the expressions share
+an indentation level. For example, an expression with an R code block
+might be:
+
+```{r}
+if (TRUE) {
+  cat("This is one expression. \n")
+  cat("This is another expression. \n")
+}
+```
+
+The equivalent in Python:
+
+```{python}
+if True:
+  print("This is one expression.")
+  print("This is another expression.")
+
+```
+
+### Container Types
```

Comment:
Would it make sense to talk about `reticulate` casting rules in this section. Ie, what happens with R lists/named lists when they are passed to python.

Reply:
Thanks, I added a sentence in the section on dictionaries about this.